pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+)
+
+type createAuthInfoOptions struct {
+	pathOptions       *pathOptions
+	name              string
+	authPath          string
+	clientCertificate string
+	clientKey         string
+	token             string
+}
+
+func NewCmdConfigSetAuthInfo(out io.Writer, pathOptions *pathOptions) *cobra.Command {
+	options := &createAuthInfoOptions{pathOptions: pathOptions}
+
+	cmd := &cobra.Command{
+		Use:   "set-user name [auth-path] [client-certificate] [client-key] [token]",
```

Comment:
Specifying optional command parameters via flag is a better user experience, imo. I'm not going to be able to remember which of the 4 args corresponds to the thing I want to set. I'd much rather type e.g.
```
set-user myself --auth-path=foo/bar/baz --token="superseecrit"
```